packages/kit/src/runtime/server/page/render.js

```diff
@@ -80,6 +81,42 @@ export async function render_response({
 			? action_result.data ?? null
 			: null;
 
+	/** @type {string} */
+	let base = paths.base;
+
+	/** @type {string} */
+	let assets = paths.assets;
+
+	/**
+	 * An expression that will evaluate in the client to determine the resolved base path.
+	 * We use a relative path when possible to support IPFS, the internet archive, etc.
+	 */
+	let base_expression = s(paths.base);
+
+	// if appropriate, use relative paths for greater portability
+	if (paths.relative !== false && !state.prerendering?.fallback) {
+		const segments = event.url.pathname.slice(paths.base.length).split('/');
+
+		if (segments.length === 1 && paths.base !== '') {
+			// if we're on `/my-base-path`, relative links need to start `./my-base-path` rather than `.`
+			base = `./${paths.base.split('/').at(-1)}`;
```

Comment:
Why split.at(-1)? What if the base path spans more than one `/`, like `/foo/bar`? We know the base has to start with a slash, so why not
```suggestion
			base = `.${paths.base}`;
```

Reply:
Consider a basepath like `/a/b/c`. If you're rendering that route, then `{base}/d` should resolve to `/a/b/c/d`.

With non-relative paths, that's easy — just use `base` as written. But if you're using a relative path, then it needs to be `./c` to produce `./c/d` to resolve to `/a/b/c/d` (or `../b/c/d`, or `../../a/b/c/d`, but there's no point in doing that).

With your suggestion, it would be `./a/b/c`, which would resolve to `/a/b/a/b/c/d`